HeatStroke/GOComponents/GameObject.cpp

```diff
@@ -113,6 +120,66 @@ void GameObject::DeleteAllComponents()
 	m_mComponentMap.clear();
 }
 
+
+void GameObject::AddChild(GameObject* p_pChild)
+{
+	if (p_pChild != nullptr)
+	{
+		m_mChildMap.insert(std::pair<std::string, GameObject*>(p_pChild->GetGUID(), p_pChild));
+		p_pChild->SetParent(this);
+		m_Transform.AddChild(&(p_pChild->GetTransform()));
```

Comment:
The call to `GameObject::RemoveChild` inside of `GameObject::SetParent` isn't being called, as `pOldParent == nullptr`, so I'm not sure if that's the problem.  I may be missing something though.